apis/supervisor/idp/v1alpha1/types_githubidentityprovider.go.tmpl

```diff
@@ -0,0 +1,218 @@
+// See the response for https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#list-teams-for-the-authenticated-user.
+type GitHubGroupNamesSpec string
+
+const (
+	// GitHubGroupsAsNames specifies using the `name` attribute from GitHub as the downstream group names
+	GitHubGroupsAsNames GitHubGroupNamesSpec = "name"
+
+	// GitHubGroupsAsSlugs specifies using the `slug` attribute from GitHub as the downstream group names
+	GitHubGroupsAsSlugs GitHubGroupNamesSpec = "slug"
+)
+
+// GitHubClaims allows customization of the username and groups claims
+type GitHubClaims struct {
+	// Username configures which property of the GitHub user record shall determine the username in Kubernetes.
+	//
+	// Can be either "id", "login", or "login:id". Defaults to "login:id".
+	//
+	// GitHub's user login attributes can only contain alphanumeric characters and non-repeating hyphens,
+	// and may not start or end with hyphens. GitHub users are allowed to change their login name,
+	// although it is inconvenient. If a GitHub user changed their login name from "foo" to "bar",
+	// then a second user might change their name from "baz" to "foo" in order to take the old
+	// username of the first user. For this reason, it is not as safe to make authorization decisions
+	// based only on the user's login attribute.
+	//
+	// If desired, an admin could configure identity transformation expressions on a FederationDomain to
+	// further customize how these usernames are presented to Kubernetes.
+	//
+	// Defaults to "login:id", which is the user login attribute, followed by a colon, followed by the unique and
+	// unchanging integer ID number attribute. This blends human-readable login names with the unchanging ID value
+	// from GitHub. Note that colons are not allowed in GitHub login attributes or ID numbers, so the colon in the
+	// "login:id" name will always be the login:id separator colon.
+	//
+	// +kubebuilder:default="login:id"
+	// +kubebuilder:validation:Enum={"id","login","login:id"}
+	Username GitHubUsernameClaimSpec `json:"username"`
+
+	// Groups configures which property of the GitHub team record shall determine the group names in Kubernetes.
+	//
+	// Can be either "name" or "slug". Defaults to "slug".
+	//
+	// GitHub team names can contain upper and lower case characters, whitespace, and punctuation (e.g. "Kube admins!").
+	//
+	// GitHub team slugs are lower case alphanumeric characters and may contain dashes (e.g. "kube-admins").
+	//
+	// Either way, group names as presented to Kubernetes will always be prefixed by the GitHub organization
+	// name followed by a forward slash (e.g. "my-org/my-team"). GitHub organization login names can only contain
+	// alphanumeric characters or single hyphens, so the first forward slash `/` will be the separator between
+	// the organization login name and the team name or slug.
+	//
+	// If desired, an admin could configure identity transformation expressions on a FederationDomain to
```

Comment:
Should we explicitly say "Pinniped FederationDomain" to make sure the user knows this isn't a GitHub concept?

Reply:
I like that. I'll go a step further to say Pinniped Supervisor FederationDomain.